specification/apimanagement/control-plane/Microsoft.ApiManagement/preview/2017-03-01/apimanagement.json

```diff
@@ -0,0 +1,484 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApiManagementClient",
+    "description": "Use these REST APIs for performing operations on entities like API, Product, and Subscription associated with your Azure API Management deployment.",
+    "version": "2017-03-01"
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{apimBaseUrl}",
+    "useSchemePrefix": false,
+    "positionInOperation": "first",
+    "parameters": [
+      {
+        "name": "apimBaseUrl",
+        "description": "The management endpoint of the API Management service, for example https://myapimservice.management.azure-api.net.",
+        "required": true,
+        "type": "string",
+        "in": "path",
+        "x-ms-skip-url-encoding": true
+      }
+    ]
+  },
```

Comment:
Have you considered to make this look like...
```json
 "x-ms-parameterized-host": {
    "hostTemplate": "{apiUrl}.{suffix}",
    "parameters": [
     {
        "name": "apiUrl",
        "required": true,
        "type": "string",
        "in": "path",
        "x-ms-skip-url-encoding": true
      },
      {
        "name": "suffix",
      "in": "path",
      "required": true,
      "type": "string",
      "default": "management.azure-api.net",
      }
    ]
  }
```

Reply:
This is a good suggestion, but it will not work for us as
1. The suffix can be multiple `management.azure-api.net`, `management.azure-api.cn`, `management.azure-api.usgov`
2. We also allow custom domain on the management endpoint, so customer can override the default `contoso.management.azure-api.net` with `management.contoso.com`

Hence having one url is fine for us.